Application/Api/Education/ClassRegister/ApiDiary.php

```diff
@@ -0,0 +1,585 @@
+use SPHERE\Common\Frontend\Link\Repository\Primary;
+use SPHERE\Common\Frontend\Link\Repository\Standard;
+use SPHERE\Common\Frontend\Message\Repository\Danger;
+use SPHERE\Common\Frontend\Message\Repository\Success;
+use SPHERE\Common\Frontend\Table\Structure\TableData;
+use SPHERE\Common\Window\Redirect;
+use SPHERE\System\Extension\Extension;
+
+/**
+ * Class ApiDiary
+ *
+ * @package SPHERE\Application\Api\Education\ClassRegister
+ */
+class ApiDiary extends Extension implements IApiInterface
+{
+
+    use ApiTrait;
+
+    /**
+     * @param string $Method
+     *
+     * @return string
+     */
+    public function exportApi($Method = '')
+    {
+        $Dispatcher = new Dispatcher(__CLASS__);
+
+        $Dispatcher->registerMethod('loadDiaryContent');
+
+        $Dispatcher->registerMethod('openCreateDiaryModal');
+        $Dispatcher->registerMethod('saveCreateDiaryModal');
+
+        $Dispatcher->registerMethod('openEditDiaryModal');
+        $Dispatcher->registerMethod('saveEditDiaryModal');
+
+        $Dispatcher->registerMethod('openDeleteDiaryModal');
+        $Dispatcher->registerMethod('saveDeleteDiaryModal');
+
+        $Dispatcher->registerMethod('openSelectStudentModal');
+        $Dispatcher->registerMethod('saveSelectStudentModal');
+
+        return $Dispatcher->callMethod($Method);
+    }
+
+    /**
+     * @return ModalReceiver
+     */
+    public static function receiverModal()
+    {
+
+        return (new ModalReceiver(null, new Close()))->setIdentifier('ModalReceiver');
+    }
+
+    /**
+     * @param string $Content
+     * @param string $Identifier
+     *
+     * @return BlockReceiver
+     */
+    public static function receiverBlock($Content = '', $Identifier = '')
+    {
+
+        return (new BlockReceiver($Content))->setIdentifier($Identifier);
+    }
+
+    /**
+     * @return Pipeline
+     */
+    public static function pipelineClose()
+    {
+        $Pipeline = new Pipeline();
+        $Pipeline->appendEmitter((new CloseModal(self::receiverModal()))->getEmitter());
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param int|null $DivisionId
+     * @param int|null $GroupId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineLoadDiaryContent($DivisionId = null, $GroupId = null)
+    {
+        $Pipeline = new Pipeline(false);
+        $ModalEmitter = new ServerEmitter(self::receiverBlock('', 'DiaryContent'), self::getEndpoint());
+        $ModalEmitter->setGetPayload(array(
+            self::API_TARGET => 'loadDiaryContent',
+        ));
+        $ModalEmitter->setPostPayload(array(
+            'DivisionId' => $DivisionId,
+            'GroupId' => $GroupId
+        ));
+        $Pipeline->appendEmitter($ModalEmitter);
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param int|null $DivisionId
+     * @param int|null $GroupId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineOpenCreateDiaryModal($DivisionId = null, $GroupId = null)
```

Comment:
Codacy found an issue: [Incorrect spacing between argument "$DivisionId" and equals sign; expected 0 but found 1](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)